pandas/_libs/parsers.pyx

```diff
@@ -1316,8 +1316,8 @@ cdef class TextReader:
         else:
             if self.header is not None:
                 j = i - self.leading_cols
-                # hack for #2442
-                if j == len(self.header[0]):
+                # hack for #2442 and #26218
```

Comment:
can you remove the 'hack' word; rather put an explan what is happening here

Reply:
Fixed